### ISSUE TITLE
Incorrect size calculation with long class names?

### ISSUE THREAD

**Issue by sinbad** [fpwong/BlueprintAssistWiki#114]
I REALLY like BPAssist, it saves me sooo much time, so thankyou for this tool.

I'm getting a few cases where nodes are overlapping, and I'm wondering if it's due to the length of the class names. I've tried selecting all the nodes and doing Ctrl+Shift+R to refresh the node size, but it doesn't help.

If you zoom out, the layout looks correct and doesn't overlap because the "Target is .." section gets hidden. As soon as you zoom in, things overlap again as the node expands with the class name:

Zoomed out:
![image](https://user-images.githubusercontent.com/142735/221171617-0ef5aff7-4728-4895-9dda-37b22b08ab69.png)

Zoomed in:
![image](https://user-images.githubusercontent.com/142735/221171680-21967675-4429-4930-89f4-47a1c2cf1100.png)

**Comment by fpwong** [comment]
Hi @sinbad, I don't think this is a size issue but something else. I did encounter something very similar a while ago but I thought I had fixed the issue.

Could you try disabling the setting `Expand Nodes by Height` and let me know if this fixed it?

It would also help greatly if you could display the other linked nodes to the left so I can test it on my side. You can copy and paste them here to easily share them: https://blueprintue.com/

**Comment by sinbad** [comment]
I went back to copy more of it, and magically it's now fixed XD I've been editing other parts of the graph so I guess something else has forced a recalc - although I'm pretty sure I selected all & pressed "F".

In case it's still useful to know here's the nodes to the left. It's quite a big graph so I don't want to post all of it, here it is looking fine now: 🤷‍♂️ 
![image](https://user-images.githubusercontent.com/142735/221179852-e7c35789-7def-48aa-b004-9fd941f138a3.png)

**Comment by fpwong** [comment]
👍

**Comment by sinbad** [comment]
Ah, I found another instance of it: https://blueprintue.com/blueprint/lsdo3x_5/

The answer in this case was to press Ctrl+Shift+R, then to force a reformat with "F", which fixed it. I could have sworn I did that in the other case but it definitely worked this time. Perhaps either the FAQ should explicitly say that you have to reformat as well as doing RefreshNodeSize, or perhaps reformat should automatically recalculate the node size (or have an option to do so)?

**Comment by fpwong** [comment]
I couldn't get this example to fail on my side. I assume it is the `Set CurrentState Enum` node which had a bad size for you in this case?

By default the plugin will detect when nodes need to refresh their size and automatically do this before formatting. Check this setting `Refresh Node Size Before Formatting` is enabled.

This is somewhat of a new feature, so there may be some cases where this will not work correctly. I have been fixed some bugs relating to this recently so also check you are on the latest version 4.1.2.

EDIT: I added a fix in 4.0.11 where I changed how it read the node's title. This would explain why you found it to be an issue on long title names!
